src/confd/yang/infix-containers@2024-02-01.yang

```diff
@@ -0,0 +1,367 @@
+
+                       With this setting the source path is not used, instead
+                       a temporary read-only file is created on the host with
+                       this content and bind-mounted into the container at the
+                       target destination path.
+
+                       In the CLI, use 'set content' without and argument to
+                       open an editor for easy copy-paste.  On exit from the
+                       editor the contents are base64 encoded automatically.";
+              type binary;
+            }
+          }
+        }
+
+        leaf target {
+          description "Absolute path to target destination inside container.
+
+                       For example, to mount file on /etc/ntpd.conf set this path and
+                       then use either the 'source' path to a file on the host system,
+                       or import the text file using the 'content' node.";
+          mandatory true;
+          type string {
+            pattern '/.*';
+          }
+        }
+
+        leaf read-only {
+          description "All mounts are read-only by default.
+                       Use this option to allow containers to write to files
+                       and directories on the host system.
+
+                       NOTE: 'content' files can also be set read-write, but
+                             no changes are saved back to hosts's datastore.";
+          type boolean;
+          default true;
+        }
+      }
+
+      list volume {
+	description "Create a writable volume that survive container upgrades.";
+	key name;
+
+	leaf name {
+	  description "Single word, [A-Za-z_], to identify this volume.
+
+                       Volumes are retained over the lifetime of a container and
+                       survive both upgrading the image and configuration changes,
+                       which otherwise wipe the default writable layer a container
+                       is given.
+
+                       Volumes combine well with 'read-only' containers, when you
+                       know which files/directories you want to persist.
+
+                       Compared to a bind mount, a volume is automatically 'synced'
+                       with the contents of the container's file system on first
+                       use.  Hence, upgrading the container image will not update
+                       the volume if the image has new/removed files at 'path'.";
+	  type string;
+	}
+
+	leaf target {
+	  description "Absolute path to target destination directory inside the container.";
+	  mandatory true;
+          type string {
+            pattern '/.*';
+          }
+	}
+      }
+
+      leaf running {
+        description "Status of container, running or not.";
+        config false;
+        type boolean;
+      }
+
+      leaf status {
+        description "Status of container, human friendly.";
+        config false;
+        type string;
```

Comment:
How many possible settings for this are there, looks like a enum is a better choise

Reply:
I tried to find out, because I too wanted an enum, but failed.

The status is taken from "podman inspect foo", and can be, e.g., "configured", "up almost 2 hours", etc.  So I resigned and just took it as a sign that we should just forward the human readable status as a string from whatever container daemon we have.

Reply:
The state and uptime seems to be two different nodes for me.

Reply:
The data we can get from `podman inspect foo` looks like this:
```json
          "State": {
               "OciVersion": "1.1.0-rc.1",
               "Status": "created",
               "Running": false,
```
The `"Status"` node is the human readable format from `podman`, which can also be `"Up 2 hours five minutes, almost"`, or any other fairy tale.  I've checked everywhere but it cannot be broken up more than what I've done.  So I decided to just take the "human readable" format and forward it in our model to the user as-is.